cpp/include/cudf/detail/utilities/hash_functions.cuh

```diff
@@ -231,6 +231,9 @@ MD5ListHasher::operator()<string_view>(column_device_view data_col,
 }
 
 struct MD5Hash {
+  MD5Hash() = default;
+  CUDA_HOST_DEVICE_CALLABLE MD5Hash(uint32_t seed) : m_seed(seed) {}
```

Comment:
Can you replace `CUDA_HOST_DEVICE_CALLABLE` with `constexpr` here? If not, does this need to be callable from both host and device?

Reply:
looks like I can. I just followed constructors of the other hash functions, which had `CUDA_HOST_DEVICE_CALLABLE`. Let me see if I can replace them all to either `constexpr` or `CUDA_DEVICE_CALLABLE`.